rotkehlchen/chain/gnosis/modules/sdai/decoder.py

```diff
@@ -0,0 +1,119 @@
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Deposit {event.balance.amount} {asset.symbol} into the sDAI contract'  # noqa: E501
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == shares
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f'Withdraw {event.balance.amount} sDAI from sDAI contract'
+                event.counterparty = CPT_SDAI
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_sdai_redeem_events(self, context: DecoderContext) -> DecodingOutput:
+        # 18 is the number of decimals for sdai
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        amount = token_normalized_value_decimals(amount_raw, 18)
+        shares_raw = hex_or_bytes_to_int(context.tx_log.data[32:64])
+        shares = token_normalized_value_decimals(shares_raw, 18)
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == shares
+            ):
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.event_type = HistoryEventType.DEPOSIT
+                event.notes = f'Return {event.balance.amount} sDAI to sDAI contract'
+                event.counterparty = CPT_SDAI
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == amount
+            ):
+                asset = event.asset.resolve_to_crypto_asset()
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f'Withdraw {event.balance.amount} {asset.symbol} from sDAI contract'
+                event.counterparty = CPT_SDAI
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def decode_sdai_events(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] == SDAI_DEPOSIT:
+            return self._decode_sdai_deposit_events(context)
+        if context.tx_log.topics[0] == SDAI_REDEEM:
+            return self._decode_sdai_redeem_events(context)
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {GNOSIS_SDAI_ADDRESS: (self.decode_sdai_events,)}
+
+    @staticmethod
+    def counterparties() -> tuple[CounterpartyDetails, ...]:
+        return (CounterpartyDetails(
+            identifier=CPT_SDAI,
+            label=SDAI_LABEL,
+            image=SDAI_ICON,
+        ),)
```

Comment:
you can create the 

```python
CounterpartyDetails(
            identifier=CPT_SDAI,
            label=SDAI_LABEL,
            image=SDAI_ICON,
        )
```

object in evm/models/sdai/constants and have it inported bothj in the maker decoder and here